### PR TITLE
Fix tests 

### PR DESCRIPTION
**Pull request by niklaslong**
Fixes the current build on master.

### CODE REVIEW

**Comment by marcoow** [comment]
Hm, why is this necessary though?

**Comment by niklaslong** [comment]
@marcoow  The tests were failing because they were date-based. For example: selecting measurements no older than a certain DateTime. When I first wrote the tests I chose a future date (not really thinking that they would break on that date - 1st of jan 2019...), because doing it programatically was unnecessarily complex.

**Comment by marcoow** [comment]
Yes, but we must be inserting the data we then assert on somewhere and if we insert it with an `udpated-at` of 2019-1-1, the test should be independent of that?

**Comment by niklaslong** [comment]
@marcoow Not in this case as the comparison is done in the app with the current DateTime.

**Comment by niklaslong** [comment]
@marcoow This should be all set 🚀

File: apps/breethe/test/support/factories.ex [hunk 1]
```diff
@@ -18,7 +18,7 @@ defmodule Breethe.Factory do
   def measurement_factory do
     %Breethe.Data.Measurement{
       parameter: "pm10",
-      measured_at: DateTime.from_naive!(~N[2019-01-01 00:00:00.000000], "Etc/UTC"),
+      measured_at: DateTime.utc_now(),
```
Comment (marcoow) [inline apps/breethe/test/support/factories.ex]: I think what we want is to pass this date in from the test so that we know exactly what it is going to be and can assert on the value.

Comment (niklaslong) [reply apps/breethe/test/support/factories.ex]: Do you think it's unclear if it is in the factory? The assumption is that any location/measurement inserted will have the current UTC. I would be more in favour of keeping the default as is and specifying in the `insert(...)` only when using a different time? Also I think this
```elixir
      measured_at =
        location.last_updated
        |> Map.put(:microsecond, {0, 0})
        |> DateTime.to_iso8601()
```
makes the information available if need be as it explicitly points to the factory and the time isn't really relevant to a high-level understanding of the test nor the main point of the tests?

File: apps/breethe_web/test/breethe_web/controllers/location_controller_test.exs [hunk 2]
```diff
@@ -12,6 +12,11 @@ defmodule BreetheWeb.LocationControllerTest do
     test "when filtering by location name" do
       location = insert(:location, measurements: [])
 
+      measured_at =
+        location.last_updated
+        |> Map.put(:microsecond, {0, 0})
+        |> DateTime.to_iso8601()
+
```
Comment (niklaslong) [inline apps/breethe_web/test/breethe_web/controllers/location_controller_test.exs]: Do you think we should isolate this in a private function (it's used in multiple tests) or does it make the tests clearer if it is explicitly there? I don't think the way we're handling time is the point of the tests and so abstracting it away might make them clearer?

File: apps/breethe/test/support/factories.ex (hunk 1 above)
Comment (marcoow) [reply apps/breethe/test/support/factories.ex]: I think the value that's declared in the factory doesn't really matter much - the thing is we shouldn't have assertions in the tests on values that are defined in the factories as that's very hard to follow. See Andy's post on the topic here: https://simplabs.com/blog/2018/12/20/factories-best-practices.html. The thing with a fixed date is that it is either in the past in which case the value would be too old and thus not be read or in the future which is something that can never happen in reality. I think what we really want is a sensible default which would be something like `Time.now - 2.hours`

File: apps/breethe_web/test/breethe_web/controllers/location_controller_test.exs (hunk 2 above)
Comment (marcoow) [reply apps/breethe_web/test/breethe_web/controllers/location_controller_test.exs]: Sure, this could be in a private function so we reduce duplication - that function should be defined in this file though so we don't split things apart and add confusion.

File: apps/breethe/test/support/factories.ex (hunk 1 above)
Comment (niklaslong) [reply apps/breethe/test/support/factories.ex]: Ok, I'll make the changes. I don't really understand why we should use `Time.now - 2.hours` rather than `Time.now` though?

Comment (marcoow) [reply apps/breethe/test/support/factories.ex]: A measurement from 2 hours ago is just more likely than one from right at the moment when you open the app - it doesn't make a big difference but you'd always aim for reasonably _"realistic"_ default values in factories.

**Comment by niklaslong** [comment]
@marcoow build is now failing because we're running elixir 1.7.1 and the `DateTime.add/4` function is only available from 1.8 onwards. Could we revert the last commit and merge the changes without shifting the time back 2 hours, update to 1.8 so we can use `add/4` and then do a follow up PR to shift the date back?

**Comment by marcoow** [comment]
Ok, sounds good

**Comment by niklaslong** [comment]
@marcoow  I've reverted the last commit - should be all set.
